src/Addons/User.php

```diff
@@ -0,0 +1,100 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Academic Free License version 3.0
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/AFL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/AFL-3.0 Academic Free License version 3.0
+ */
+
+namespace PrestaShop\Module\Mbo\Addons;
+
+use PhpEncryptionCore as PhpEncryption;
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * This class will provide data from Addons API
+ */
+class User implements UserInterface
+{
+    /**
+     * @var PhpEncryption
+     */
+    private $encryption;
+
+    /**
+     * @var Request
+     */
+    private $request;
+
+    public function __construct()
+    {
+        $this->encryption = new PhpEncryption(_NEW_COOKIE_KEY_);
+        $this->request = Request::createFromGlobals();
```

Comment:
I don't think it's relevant to do it.
Addon User aims to be used only in HTTP Context, because it's related to the addons API, nothing else. But, you don't need the request object, you need to Cookie only which will avoid errors with subrequests, multiple request, etc.

For your information, using `Request::createFromGlobals();` doesn't mean it's usable only in Web context, it create Parameter Bags depending on PHP globals vars.